MementoMori/MementoMori.Server/Controllers/DeckController.cs

```diff
@@ -79,78 +83,98 @@ public async Task<ActionResult> EditorViewAsync(Guid deckId)
         }
 
         [HttpGet("cards")]
-        public async Task<ActionResult> GetCards(Guid deckId)
+        public IActionResult GetDueCards(Guid deckId)
         {
 
-            if (deckId == Guid.Empty)
+            try
             {
-                return BadRequest(new { errorCode = ErrorCode.InvalidInput, message = "Invalid deck ID." });
-            }
+                Guid? userId = _authService.GetRequesterId(HttpContext);
 
-            var deck = (await _deckHelper.Filter(ids: [deckId])).First();
+                if (deckId == Guid.Empty || userId == Guid.Empty)
+                {
+                    return BadRequest(new { errorCode = "InvalidInput", message = "Invalid deck or user ID." });
+                }
 
-            if (deck == null)
-                return NotFound("Deck not found.");
+                List<Card> dueForReviewCards = _cardService.GetCardsForReview(deckId, userId.Value);
 
+                if (!dueForReviewCards.Any())
+                {
+                    return NotFound("No cards due for review.");
+                }
 
-            var Cards = deck.Cards.Select(Card => new CardDTO
+                var dueCardDtos = dueForReviewCards.Select(c => new CardDTO
+                {
+                    Id = c.Id,
+                    Question = c.Question,
+                    Description = c.Description,
+                    Answer = c.Answer
+                }).ToList();
+                
+                return Ok(dueCardDtos);
+            }
+            catch (Exception ex)
             {
-                Id = Card.Id,
-                Question = Card.Question,
-                Description = Card.Description,
-                Answer = Card.Answer,
-
-            }).ToList();
+                Console.Error.WriteLine($"Error in GetDueCards: {ex.Message}\n{ex.StackTrace}");
+                Console.Error.WriteLine($"DeckId: {deckId}, UserId: {_authService.GetRequesterId(HttpContext)}");
+                return StatusCode(500, new { errorCode = "ServerError", message = "An unexpected error occurred." });
+            }
+        }
+        [HttpPost("addToCollection")]
+        public IActionResult AddCardsToCollection(Guid deckId)
+        {
+            Guid? userId = _authService.GetRequesterId(HttpContext);
 
-            return Ok(Cards);
+            if(deckId == Guid.Empty || userId == Guid.Empty || userId == null )
```

Comment:
A user created guid will never be Empty due to how a guid is created so no need to check for it.

Reply:
No problem checking it though, in case a user who is not logged in gets in practice page.